### ISSUE TITLE
verify gml files after they are created

### ISSUE THREAD

**Issue by agilliland** [opencitymodel/data-pipeline#26]
When we are generating citygml files we should add a quick verification step which opens up the newly written citygml files and attempts to verify them before moving on to the next file as sometimes files can end up being corrupted.

**Comment by clausnagel** [comment]
Just curious, why are the files sometimes corrupted?

**Comment by clausnagel** [comment]
Thanks for explaining. Saw that you are using citygml4j and thought that maybe it causes the issue. In this case I possibly could have been of help.

**Comment by clausnagel** [comment]
Checked your `CitygmlBuilder` and saw that you are creating 40k buildings in main memory which are then written to a file. 40k LOD1 buildings should not occupy much main memory though. 

citygml4j supports writing chunks. You could create a building, send it to the file and remove it afterwards. This helps to keep the memory footprint low and works for both GML and CityJSON. However, in CityJSON, city objects are not self-contained due to global properties like the `"vertices"` array. So, even when writing buildings chunk-wise, some information must be kept in main memory until the file is closed. Nevertheless, the memory footprint should still be substantially lower compared to creating all buildings in main memory.

Using chunk-wise writing would only require few changes to your code.

**Comment by agilliland** [comment]
Yeah, that would be really helpful!  I've definitely seen the CityJSON creation take up much more memory than I had expected and given the way the vertices are managed that makes sense, but anything to keep the memory footprint down a bit more would be welcome.

**Comment by clausnagel** [comment]
I'm happy to work on an example how it could look like.
